lib/src/rules/avoid_relative_imports.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r'Avoid relative imports for all files`.';
+
+const _details = r'''*DO* avoid relative imports for files in `lib/`.
+
+Mixing relative and absolute imports makes it complicated to refactor across
+multiple packages.
+
+This is a more strict version of 'avoid_relative_lib_imports'.
```

Comment:
What this rule doesn't do (which avoid_relative_lib_imports does implicitly) is to guard against a false positive when a library outside the `lib` directory imports another library outside the `lib` directory. This happens, for example, in some test code. In those cases a relative path is all that can be used to reference the library.